crates/release_plz_core/src/cargo.rs

```diff
@@ -68,6 +79,17 @@ pub fn is_published(index: &mut Index, package: &Package) -> anyhow::Result<bool
     Ok(is_in_cache(index, package))
 }
 
+pub async fn is_published_sparse(index: &SparseIndex, package: &Package) -> anyhow::Result<bool> {
+    let url = index.crate_url(&package.name);
+
+    match url {
+        // sparse protocol can fetch the manifest for a single crate, so we can
+        // hit the registry directly
+        Some(u) => Ok(reqwest::get(u).await?.status().is_success()),
+        None => Ok(false),
+    }
+}
```

Comment:
in this function, shouldn't we verify the package version, too?

Reply:
You're correct.  I just added that.